### ISSUE TITLE
Cannot fine-tune your model with default arguments

### ISSUE THREAD

**Issue by ftakanashi** [facebookresearch/Mask-Predict#4]
Hello. I download your en-zh trained model and try to fine-tune it with my own data (also part of WMT17 en-zh corpus). But when I run the train.py script, I encountered the following error:
```
-- Process 0 terminated with the following error:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/torch/multiprocessing/spawn.py", line 19, in _wrap
    fn(i, *args)
  File "/root/work/Mask-Predict/train.py", line 265, in distributed_main
    main(args, init_distributed=True)
  File "/root/work/Mask-Predict/train.py", line 80, in main
    train(args, trainer, task, epoch_itr)
  File "/root/work/Mask-Predict/train.py", line 121, in train
    log_output = trainer.train_step(samples)
  File "/root/work/Mask-Predict/fairseq/trainer.py", line 347, in train_step
    self.optimizer.step()
  File "/root/work/Mask-Predict/fairseq/optim/fp16_optimizer.py", line 197, in step
    self.fp32_optimizer.step(closure)
  File "/root/work/Mask-Predict/fairseq/optim/fairseq_optimizer.py", line 94, in step
    self.optimizer.step(closure)
  File "/root/work/Mask-Predict/fairseq/optim/adam.py", line 114, in step
    amsgrad = group['amsgrad']
KeyError: 'amsgrad'
```

My arguments for train.py are:
```
python train.py {lang_prefix}/data-bin --arch bert_transformer_seq2seq --criterion label_smoothed_length_cross_entropy --label-smoothing 0.1 --lr 5e-4 --warmup-init-lr 1e-7 --min-lr 1e-9 --lr-scheduler inverse_sqrt --warmup-updates 10000 --optimizer adam --adam-betas '(0.9, 0.999)' --adam-eps 1e-6 --task translation_self --max-tokens 1024 --weight-decay 0.01 --dropout 0.3 --encoder-layers 6 --encoder-embed-dim 512 --decoder-layers 6 --decoder-embed-dim 512 --max-source-positions 10000 --max-target-positions 10000 --fp16 --max-update 300000 --seed 0 --save-dir {lang_prefix}/model
```
You can see that it is almost identical to the default ones. Are there any subtle differences between your default arguments and released pre-trained model?

**Comment by yinhanliu** [comment]
Can you try it with --reset-optimizer?

**Comment by ftakanashi** [comment]
> Can you try it with --reset-optimizer?

It works with --reset-optimizer. Thx!

**Comment by SunbowLiu** [comment]
> > Can you try it with --reset-optimizer?
> 
> It works with --reset-optimizer. Thx!

Hi,

I am curious about the fine-tuned performance, have you get any improvement?

**Comment by ftakanashi** [comment]
> > > Can you try it with --reset-optimizer?
> > 
> > 
> > It works with --reset-optimizer. Thx!
> 
> Hi,
> 
> I am curious about the fine-tuned performance, have you get any improvement?

Hello,

Umm, actually now I think what I had done is more like continuous-training rather than fine-tuning...
Anyway, I used the same WMT17-En-Zh dataset mentioned in the paper and trained the released en-zh model for another extra 10 epochs.
The result is not very satisifying because final BLEU score by the extra-trained model is 25.22(T=4) 
 while the released one has a 32.63(T=4) as mentioned in the paper.
